caffe2/utils/threadpool/pthreadpool-cpp.cc

```diff
@@ -0,0 +1,75 @@
+#include <caffe2/utils/threadpool/pthreadpool-cpp.h>
+#include <c10/util/Exception.h>
+
+namespace caffe2 {
+
+PThreadPool::PThreadPool(const size_t thread_count)
+  : threadpool_(pthreadpool_create(thread_count), pthreadpool_destroy) {
+}
+
+size_t PThreadPool::get_thread_count() const {
+  TORCH_INTERNAL_ASSERT(threadpool_.get(), "Invalid threadpool!");
+  return pthreadpool_get_threads_count(threadpool_.get());
+}
+
+void PThreadPool::set_thread_count(const size_t thread_count) {
+  // As it stands, pthreadpool is an entirely data parallel framework with no
+  // support for task parallelism.  Hence, all functions are blocking, and no
+  // user-provided tasks can be in flight when the control is returned to the
+  // user of the API, which means re-initializing the library, without the
+  // need to wait on any pending tasks, is all one needs to do to re-adjust
+  // the thread count.
```

Comment:
Thanks for the comment.

Reply:
What if one thread calls `set_thread_count` while another thread is running a model?